File: cpp/LightWaterReactor1G.cpp
// One-Group Light Water Reactor Component Class

#include "LightWaterReactor1G.h"
LightWaterReactor1G::LightWaterReactor1G() : Reactor1G(lwr_defaults, lwr_p2track)
{
};

LightWaterReactor1G::LightWaterReactor1G(std::string h5lib, std::string n) : Reactor1G(lwr_defaults, lwr_p2track, n)
{
    libfile = h5lib;
    loadlib(h5lib);
};

LightWaterReactor1G::LightWaterReactor1G(ReactorParameters rp, std::string n) : Reactor1G(rp, lwr_p2track, n)
{
};

LightWaterReactor1G::LightWaterReactor1G(std::string h5lib, ReactorParameters rp, std::string n) : Reactor1G(rp, lwr_p2track, n)
{
    libfile = h5lib;
    loadlib(h5lib);
};

LightWaterReactor1G::~LightWaterReactor1G() 
{
};

void LightWaterReactor1G::calc_params()
{
    /** Sets relevent LWR parameters.
     *  Overwrites standard, do-nothing calc_params() function.
     */

    calcSubStreams();

    params_prior_calc["BUd"]  = 0.0;
    params_after_calc["BUd"] = BUd;

    params_prior_calc["U"]  = mat_feed_u.mass;
    params_after_calc["U"] = mat_prod_u.mass;

    params_prior_calc["TRU"]  = mat_feed_tru.mass;
    params_after_calc["TRU"] = mat_prod_tru.mass;

    params_prior_calc["ACT"]  = mat_feed_act.mass;
    params_after_calc["ACT"] = mat_prod_act.mass;

    params_prior_calc["LAN"]  = mat_feed_lan.mass;
    params_after_calc["LAN"] = mat_prod_lan.mass;

    params_prior_calc["FP"]  = 1.0 - mat_feed_act.mass  - mat_feed_lan.mass;
    params_after_calc["FP"] = 1.0 - mat_prod_act.mass - mat_prod_lan.mass;

    return;
};

File: cpp/light_water_reactor1g.py
class_ds = \
"""A One-Group Light Water Reactor Fuel Cycle Component.  This is a daughter 
class of Reactor1G and a granddaughter of FCComp.

Parameters
----------
lib : str, optional
    The path the the LWR HDF5 data library.  This value is set to 
    Reactor1G.libfile and used by Reactor1G.loadlib().
rp : ReactorParameters, optional
    The physical reactor parameter data to initialize this LWR instance with.  
    If this argument is not provided, default values are taken.
n : str, optional
    The name of this LWR instance.

"""

desc = {
    'docstrings': {
        'class': class_ds,
        'attrs': {},
        'methods': {},
        },
    'attrs': {},
    'extra': {},
    }

mod = {'LightWaterReactor1G': desc,
       'docstring': "Python wrapper for LWR1G.",}


desc['docstrings']['methods']['calc_params'] = \
"""Along with its own parameter set to track, the LWR model implements its own 
function to set these parameters.  This function is equivalent to the following::

    self.params_prior_calc["BUd"]  = 0.0
    self.params_after_calc["BUd"] = self.BUd

    self.params_prior_calc["U"]  = self.mat_feed_u.mass
    self.params_after_calc["U"] = self.mat_prod_u.mass

    self.params_prior_calc["TRU"]  = self.mat_feed_tru.mass
    self.params_after_calc["TRU"] = self.mat_prod_tru.mass

    self.params_prior_calc["ACT"]  = self.mat_feed_act.mass
    self.params_after_calc["ACT"] = self.mat_prod_act.mass

    self.params_prior_calc["LAN"]  = self.mat_feed_lan.mass
    self.params_after_calc["LAN"] = self.mat_prod_lan.mass

    self.params_prior_calc["FP"]  = 1.0 - self.mat_feed_act.mass  - self.mat_feed_lan.mass

"""
